async.h

```diff
@@ -42,9 +42,11 @@ struct dict; /* dictionary header is included in async.c */
 
 /* Reply callback prototype and container */
 typedef void (redisCallbackFn)(struct redisAsyncContext*, void*, void*);
+typedef void (redisFinalizerCallback)(struct redisAsyncContext*, void*);
 typedef struct redisCallback {
     struct redisCallback *next; /* simple singly linked list */
     redisCallbackFn *fn;
+    redisFinalizerCallback *finalizer;
     int pending_subs;
     void *privdata;
```

Comment:
I think we have to put the new member at the end of the structure,  as the current changes breaks ABI compatibility by inserting a member in the middle.

```suggestion
    int pending_subs;
    void *privdata;
    redisFinalizerCallback *finalizer;
```

![Screen Shot 2021-08-08 at 7 09 28 PM](https://user-images.githubusercontent.com/468149/128653963-d6b5e877-dcac-4690-ac80-a90e83287c55.png)